plugin/src/main/java/org/opensearch/ml/action/agents/GetAgentTransportAction.java

```diff
@@ -80,6 +80,9 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLAgen
         MLAgentGetRequest mlAgentGetRequest = MLAgentGetRequest.fromActionRequest(request);
         String agentId = mlAgentGetRequest.getAgentId();
         String tenantId = mlAgentGetRequest.getTenantId();
+        if (!TenantAwareHelper.validateTenantId(mlFeatureEnabledSetting, tenantId, actionListener)) {
+            return;
```

Comment:
We will need a similar validation for agent execute action.

Reply:
Addressed : https://github.com/opensearch-project/ml-commons/blob/ee1181c93de55a7c0c1509586e07c0aeebcbfe20/ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLAgentExecutor.java#L120-L122